### ISSUE TITLE
Clear BTC wallet data after wallet switches

### ISSUE THREAD

**Issue by alex-miller-0** [GridPlus/lattice-manager#187]
When switching from SafeCard <-> Lattice it doesn't look like the wallet data is automatically cleared. The user needs to hit the refresh button to load the wallet data for the new interface. This should not be the case.

**Comment by douglance** [comment]
Would it make sense to add a poll every few seconds in the app to check for changes in the wallet status?

**Comment by alex-miller-0** [comment]
Definitely not every few seconds because we will get rate limited. Depending on the number of addresses, a user may need to make 5+ requests with each state refresh. I think we could do it every couple minutes though... I just wanted to take the refreshing logic out to reduce complexity when I was migrating to this new architecture

**Comment by douglance** [comment]
Right--I didn't realize it was talking to the server. For some reason I thought it was direct to the Lattice. 

If I understand correctly, in a future update, this could be done with WebSockets to improve the user experience. We could push the update to the client, rather than wait for a request.

**Comment by alex-miller-0** [comment]
Not sure I follow re:websockets but yeah we could make the page more responsive generally... it just adds more complexity and the BTC wallet is not a widely used feature (as evidenced by the fact that we hide it by default lol) so I'd prefer we not spend a ton of time on it with higher priority things elsewhere

**Comment by alex-miller-0** [comment]
I just tested this out again and can be a bit more specific:

1. Started manager with a wallet that has BTC history. Went to BTC wallet and pressed `Start`, which synced my addresses + history
2. Removed SafeCard and clicked on the `SafeCard` button in the header. This caused the wallet to refresh (`refreshWallets` (`main.js`) -> `refreshWallets` (`sdkSession.js`)
3. The screen now shows empty BTC history, but it didn't take any time to update addresses. I had to click `Refresh` inside of the wallet module to do that

Looking at the code I actually thing this issue is a one line fix. `refreshWallets` in `sdkSession.js` calls `this.getBtcWalletData`, but it needs to first call `fetchBtcAddresses` and only call `getBtcWalletData` in the callback of that function.
